src/TraceEvent/Computers/ThreadTimeComputer.cs

```diff
@@ -648,6 +709,87 @@ private void OnSampledProfile(TraceEvent data)
             }
         }
 
+        private void OnUniversalCSwitch(SampleTraceData data)
+        {
+            // This event fires when a thread is switched back in.
+            // The timestamp is the time at which the thread was switched back in.
+            // The value is the amount of time that thread was switched out.
+
+            TraceThread thread = data.Thread();
+            if (thread == null)
+            {
+                Debug.WriteLine("Warning, no thread at " + data.TimeStampRelativeMSec.ToString("f3"));
+                return;
+            }
+
+            // Get the blocking stack.
+            StackSourceCallStackIndex stackIndex = GetCallStack(data, thread);
+
+            // If the thread wasn't previously marked as blocked, then mark it.
+            if (!m_threadState[(int)thread.ThreadIndex].ThreadBlocked)
+            {
+                double startTimeStampRelMsec = m_eventLog.QPCTimeToRelMSec(data.TimeStampQPC - (long)data.Value);
```

Comment:
Below, you use data.TimeStampRelativeMSec. This is calculation different? The answer to this may answer my question above